crates/ra_ide/src/hover.rs

```diff
@@ -1310,4 +1369,1045 @@ fn func(foo: i32) { if true { <|>foo; }; }
             ]
             "###);
     }
+
+    #[test]
+    fn test_hover_struct_has_goto_type_action() {
+        let (_, actions) = check_hover_result(
+            "
+            //- /main.rs
+            struct S{ f1: u32 }
+
+            fn main() {
+                let s<|>t = S{ f1:0 };
+            }
+            ",
+            &["S"],
+        );
+        assert_debug_snapshot!(actions,
```

Comment:
In a separate PR, it might be a good idea to try to make hover tests more dense. The `derive(Debug)` is a lazy way to get something testable, but I think with a targeted manual impl we can have much shorter expected output.